src/components/Button/Button.tsx

```diff
@@ -1,49 +1,186 @@
-import React, { ButtonHTMLAttributes, PropsWithChildren } from 'react';
+import * as React from 'react';
+import { Slot } from '@radix-ui/react-slot';
+import { cva, type VariantProps } from 'class-variance-authority';
 import { cn } from '../../helpers/classnames';
+import { Spinner } from '../spinner';
 
-export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
-  onClick?: () => void;
-  tooltip?: string;
-  filled?: boolean;
-  ghost?: boolean;
-  red?: true;
+const buttonVariants = cva(
```

Comment:
this might be more of an experiment than an actual solution, but perhaps the hover/active/disabled states can be at least mostly done via the `after` pseudo-element.

Think something like `className="relative bg-green-200 after:content-[''] after:absolute after:inset-0 after:bg-black after:opacity-0 hover:after:opacity-20 active:after:opacity-30"`. Combined with maybe setting mix-blend-mode (https://tailwindcss.com/docs/mix-blend-mode), especially to remove saturation on disabled.

By doing this we might not need as many variant-unique classes, since you don't need custom hover/active/disabled classes for each variant / color.

The above is a non-tested solution and would require some more work